bundles/org.openhab.binding.generacmobilelink/src/main/java/org/openhab/binding/generacmobilelink/internal/handler/GeneracMobileLinkAccountHandler.java

```diff
@@ -0,0 +1,267 @@
+            if (generatorOpt.isPresent()) {
+                ((GeneracMobileLinkGeneratorHandler) childHandler).updateGeneratorStatus(generatorOpt.get());
+            }
+        }
+    }
+
+    private void stopPoll() {
+        Future<?> localPollFuture = pollFuture;
+        if (localPollFuture != null) {
+            localPollFuture.cancel(true);
+        }
+    }
+
+    private void restartPoll() {
+        stopPoll();
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 0, refreshIntervalSeconds, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        if (authToken == null) {
+            logger.debug("Attempting Login");
+            login();
+        }
+        getStatuses(true);
+    }
+
+    private synchronized void login() {
+        GeneracMobileLinkAccountConfiguration config = getConfigAs(GeneracMobileLinkAccountConfiguration.class);
+        refreshIntervalSeconds = config.refreshInterval;
+        try {
+            HTTPResult result = sendRequest(BASE_URL + "/Users/login", HttpMethod.POST, null,
+                    new StringContentProvider(
+                            gson.toJson(new LoginRequestDTO(SHARED_KEY, config.username, config.password))),
+                    "application/json");
+            if (result.responseCode == HttpStatus.OK_200) {
+                LoginResponseDTO loginResponse = gson.fromJson(result.content, LoginResponseDTO.class);
+                if (loginResponse != null) {
+                    authToken = loginResponse.authToken;
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                handleErrorResponse(result);
+                if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+                    // bad credentials, stop trying to login
+                    stopPoll();
+                }
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private void getStatuses(boolean retry) {
+        if (authToken == null) {
+            return;
+        }
+        try {
+            HTTPResult result = sendRequest(BASE_URL + "/Generator/GeneratorStatus", HttpMethod.GET, authToken, null,
+                    null);
+            if (result.responseCode == HttpStatus.OK_200) {
+                generators = gson.fromJson(result.content, GeneratorStatusResponseDTO.class);
+                updateGeneratorThings();
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                if (retry) {
+                    logger.debug("Retrying status request");
+                    getStatuses(false);
+                } else {
+                    handleErrorResponse(result);
+                }
+            }
+        } catch (InterruptedException e) {
+        }
```

Comment:
You should silently consume InterruptedExceptions. If this thread is interrupted it usually means that the system is requesting that it stops run as soon as it is able. To prevent the callers of your method from assuming that execution can continue as normal you should have your method throw the InterruptedException instead.

Let the InterruptedException travel as high up the call stack as possible to guarantee an expedient thread completion.